decode.go

```diff
@@ -1184,3 +1285,61 @@ func findEntryByKey(cur *yaml.Node, key string) (*yaml.Node, *yaml.Node, bool) {
 	}
 	return nil, cur, false
 }
+
+var nanosPerSecond = new(big.Int).SetUint64(uint64(time.Second / time.Nanosecond))
+
+var nanosMap = map[string]*big.Int{
+	"ns": new(big.Int).SetUint64(1),
```

Comment:
Just to be consistent with the rest of the lookups:
```suggestion
	"ns": new(big.Int).SetUint64(time.Nanosceond),
```

Reply:
This is the map to nanoseconds, it converts all the other units to nanos. It just happens to be that time.Nanoseconds is 1, to be consistent this would be:
```
new(big.Int).SetUint64(uint64(time.Nanosecond / time.Nanosecond)),
```

Reply:
I know, but the random magic 1 takes away from the implied documentation. What does the linter complain about?

Reply:
I'd also suggest dropping the `/ time.Nanosecond` as well, since it's also redundant if you know about the type.

Reply:
Ya, but this conde isn't supposed to know that

Reply:
I've added docs to clarify what these values represent (which is the number of nanos in each unit)

Reply:
Go's not going to change the unit value to pico's until at least Go2 XD